### PR TITLE
Get cluster gh

### PR DESCRIPTION
**Pull request by alexbleotu**
### What does this PR do?

Added utils.vmware.get_cluster
- retrieves a cluster in a datacenter based on its name

### Tests written?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: tests/unit/utils/vmware_test/cluster_test.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        self.assertEqual(excinfo.exception.strerror,
+                         'Cluster \'fake_cluster\' was not found in '
+                         'datacenter \'fake_dc\'')
+
+    def test_cluster_found(self):
+        with patch('salt.utils.vmware.get_managed_object_name',
+                   MagicMock(return_value='fake_dc')):
+            with patch('salt.utils.vmware.get_mors_with_properties',
+                       MagicMock(return_value=self.mock_entries)):
+                    res = vmware.get_cluster(self.mock_dc, 'fake_cluster2')
```
Comment (cachedout) [inline tests/unit/utils/vmware_test/cluster_test.py]: This indentation looks like it might be wrong.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by alexbleotu** [comment]
Yep saw that. Will fix when I get in the office.

**Comment by alexbleotu** [comment]
All fixed

**Comment by alexbleotu** [comment]
@cachedout You were right, the timeout increase did not fix the zeromq test... It needs more investigation

**Comment by alexbleotu** [comment]
any further issues?

**Comment by alexbleotu** [comment]
@rallytime Tests failing nothing to do with my code. Can you please merge it in so I can rebase the other pull request

**Pull request merged**